SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -799,33 +813,33 @@ bool USpatialSender::SendRPC(const FPendingRPCParams& Params)
 		Worker_ComponentId ComponentId = SchemaComponentTypeToWorkerComponentId(RPCInfo.Type);
 
 		const UObject* UnresolvedObject = nullptr;
-		Worker_CommandRequest CommandRequest = CreateRPCCommandRequest(TargetObject, Params.Payload, ComponentId, RPCInfo.Index, EntityId, UnresolvedObject);
+		Worker_CommandRequest CommandRequest = CreateRPCCommandRequest(TargetObject, Payload, ComponentId, RPCInfo.Index, EntityId, UnresolvedObject);
 
 		if (UnresolvedObject)
 		{
-			return false;
+			return ERPCError::UnresolvedTargetObject;
```

Comment:
I believe this is actually `UnresolvedParameters`

Reply:
CreateRPCCommandRequest has only this assignment "OutUnresolvedObject = TargetObject;" so it must be TargetObject. But I should probably change the CreateRPCCommandRequest function's signature to make it explicit.

Reply:
You're right. It looks like the check in `CreateRPCCommandRequest` is wrong though. It should be checking that the entity ID in that object ref is valid, not just that it resolves to an object ref. A simple case where it would go wrong is calling a CrossServer RPC on a stably named object that can be resolved but the entity hasn't been checked out so the object ref can't be used to send the command.

Reply:
Discussed it offline, resolving.